src/main/java/com/learning/mfscreener/service/XIRRCalculatorService.java

```diff
@@ -40,10 +40,7 @@ public XIRRCalculatorService(
 
     // method to calculate the total XIRR for a given PAN number
     public List<XIRRResponse> calculateTotalXIRRByPan(String pan, LocalDate asOfDate) {
-        if (asOfDate == null) {
-            asOfDate = LocalDate.now();
-        }
-        return calculateXIRRForAllFundsByPAN(pan, LocalDateUtility.getAdjustedDate(asOfDate));
+        return calculateXIRRForAllFundsByPAN(pan, LocalDateUtility.adjustDate(asOfDate));
```

Comment:
@coderabbitai, suggest code changes without breaking functionality to make more clear in this flow